langserver/langserver.py

```diff
@@ -624,6 +645,25 @@ def handle(self):
         s = LangServer(conn)
         s.run()
 
+def get_pip_package(module_path):
+    best_pkg, best_level = None, -1
+    for pkg in get_installed_distributions():
+        l = get_ancestor_level(pkg.location, module_path)
+        if l >= 0 and (best_level == -1 or l < best_level):
+            best_pkg, best_level = pkg, l
+    return best_pkg
+
+def get_ancestor_level(ancestor, child):
+    if not child.startswith(ancestor):
+        return -1
+    ancestor_cmps = ancestor.split(os.sep)
+    child_cmps = child.split(os.sep)
+    if len(ancestor_cmps) > len(child_cmps):
+        return -1
+    for i in range(0, len(ancestor_cmps)):
+        if ancestor_cmps[i] != child_cmps[i]:
+            return -1
```

Comment:
You have already established that `len(ancestor_cmps) <= len(child_cmps)`. So the pythonic thing to do here is just `ancestor_cmps == child_cmps[:len(ancestor_cmps)]`